video/out/vo_gpu_next.c

```diff
@@ -1109,8 +1109,8 @@ static void video_screenshot(struct vo *vo, struct voctrl_screenshot *args)
 
     // Passing an interpolation radius of 0 guarantees that the first frame in
     // the resulting mix is the correct frame for this PTS
-    struct pl_frame *image = (struct pl_frame *) mix.frames[0];
-    struct mp_image *mpi = image->user_data;
+    struct pl_frame image = *(struct pl_frame *) mix.frames[0];
```

Comment:
Are there any stack-size concerns to stack-allocating this struct?

Reply:
Possibly. Note that we already have a `struct pl_frame` on the stack (`target`). `sizeof(struct pl_frame)` is 616 on my system, currently.

It's also worth pointing out that `pl_render_frame` has stack that's at least `1560` bytes big, too, more in some cases. So we're easily using a few MB of stack space here. Given the limit of 8 MB even on "sane" systems, we might want to try avoiding this.

Reply:
Err, wait. I was being incredibly stupid and for some reason thinking this was in kilobytes, not bytes.

Let me retract my statement about this being problematic. We're only storing a few *kilobytes* on the stack. Not a few megabytes. I think that alleviates this worry?

Reply:
Yea, I wasn't sure if `struct pl_frame` contained a frame of image data or not. If its size is only in the order of kilobytes that should be no issue.